backend/app/Models/BatchStep.php

```
@@ -172,6 +182,38 @@ public function isBlockedByDocuments(): bool
+    }
+
+    /**
+     * Record that the given user has read and acknowledged this step's critical
+     * instructions. Idempotent and race-safe: the conditional update only fires
+     * when nothing has been recorded yet, so two concurrent acknowledgements
+     * can't overwrite each other's audit stamp. Mirrors
+     * BatchStepDocument::markValidated().
+     */
+    public function markReadConfirmed(User $user): void
+    {
+        static::whereKey($this->getKey())
```

PHPStan (2.2.2)

[error] Call to an undefined static method App\Models\BatchStep::whereKey().


(staticMethod.notFound)